pkg/vmprovider/providers/vsphere2/vmprovider_vm_test.go

```diff
@@ -56,7 +58,11 @@ func vmTests() {
 	)
 
 	BeforeEach(func() {
-		testConfig = builder.VCSimTestConfig{WithV1A2: true}
+		testConfig = builder.VCSimTestConfig{
+			WithV1A2:                  true,
+			WithContentLibrary:        true,
+			WithAutoVADPBackupRestore: true,
```

Comment:
Nit: 
Can we create a separate subsection with a "WithAutoVADPBackupRestore: enabled" for a Basic VM/TKGS VM creation. This way we can test for `with and without FSS enabled` for the presence/absence of the extraConfigKey. We can then piggyback on existing tests after FSS is enabled on main.

Reply:
I think this should be included in the default basic VM tests because AutoVADPBackupRestore will be enabled with v1a2. Adding a separate block without the B/R FSS enabled in `vsphere2/vmprovider_vm_test.go` seems unnecessary and would create additional work when we remove the FSS in the future.

Reply:
I guess it's okay cause we're close to enabling on main but I would generally prefer keeping tests for both scenarios around to have code coverage with/without FSS enabled.. v1a2 is enabled up top here because the vmprovider/vsphere2/ dir only handles v1a2 VMs. 

I get your point though. Resolving..